frontend/src/components/FAutocomplete/FAutocomplete.tsx

```diff
@@ -0,0 +1,83 @@
+import { Autocomplete } from '@mui/lab';
+import { AutocompleteInputChangeReason, TextField } from '@mui/material';
+import { useField, useFormikContext } from 'formik';
+import { useMemo } from 'react';
+
+export interface FAutocompleteOption {
+  label: string;
+  id: number;
+}
+interface FAutocompleteProps {
+  onInputChange: (
+    newValue: string | undefined,
+    reason: AutocompleteInputChangeReason,
+  ) => void;
+  inputValue: string;
+  options: FAutocompleteOption[];
+  loading: boolean;
+  label: string;
+  name: string;
+  id: string;
+  required: boolean;
+}
+export const FAutocomplete = ({
+  onInputChange: setExternalValue,
+  options,
+  loading,
+  label,
+  name,
+  required,
+}: FAutocompleteProps) => {
+  const [field, meta] = useField<number | undefined>(name);
+  const hasError = Boolean(meta.error && meta.touched);
+  const helperText = hasError ? meta.error : '';
+  const { setFieldValue } = useFormikContext();
+
+  const handleChange = (
+    e: any,
+    value: FAutocompleteOption | FAutocompleteOption[] | null,
+  ) => {
+    if (value === null) {
+      setFieldValue(name, null);
+    } else if (Array.isArray(value)) {
+      setFieldValue(name, value[0]?.id ?? null);
+    } else {
+      setFieldValue(name, value.id);
+    }
+  };
+
+  const optionsMap = useMemo(
+    () =>
+      Object.fromEntries(
+        options.map((option) => [option.id, option]),
+      ) as Record<number, FAutocompleteOption>,
+    [options],
+  );
+
+  return (
+    <Autocomplete
+      renderInput={(params) => (
+        <TextField
+          {...params}
+          required={required}
+          error={hasError}
+          label={label}
+          helperText={helperText}
+        />
+      )}
+      {...field}
+      noOptionsText="Brak wyników"
```

Comment:
When the input is empty, we should make it something like "Wpisz by wyszukać..."